resourcemanager/recaser/registration.go

```diff
@@ -0,0 +1,36 @@
+package recaser
+
+import (
+	"strings"
+	"sync"
+
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/commonids"
+	"github.com/hashicorp/go-azure-helpers/resourcemanager/resourceids"
+)
+
+var knownResourceIds = make(map[string]resourceids.ResourceId)
+
+var resourceIdsWriteLock = &sync.Mutex{}
+
+func init() {
+	//register common ids
+	resourceIdsWriteLock.Lock()
+	for _, id := range commonids.CommonIds() {
+		key := strings.ToLower(id.ID())
+		if _, ok := knownResourceIds[key]; !ok {
+			knownResourceIds[key] = id
+		}
+	}
+	resourceIdsWriteLock.Unlock()
```

Comment:
perhaps pedantic, but is there a reason not to call `RegisterResourceId` here instead?

```suggestion
	for _, id := range commonids.CommonIds() {
		RegisterResourceId(id)
	}
```

Reply:
makes sense 👍 fixed now!